### PR TITLE
Create Address repository

### PR DESCRIPTION
**Pull request by Excession27**

### CODE REVIEW

**Comment by Excession27** [comment]
Changed a typo in EmployeeDetailsRepository

File: CSharpBasics/CSB.Repository/Interfaces/IAddressRepository.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CSB.Repository.Entities;
+
+namespace CSB.Repository.Interfaces
+{
+    public interface IAddressRepository
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Interfaces/IAddressRepository.cs]: Nasledi generic repo

File: CSharpBasics/CSB.Repository/Impl/AddressRepository.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CSB.Repository.Entities;
+using CSB.Repository.Interfaces;
+
+namespace CSB.Repository.Impl
+{
+    internal class AddressRepository : IAddressRepository
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Impl/AddressRepository.cs]: Nasledi generic repo

File: CSharpBasics/CSB.Repository/Impl/AddressRepository.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        private FileDbProvider _fileDb;
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Impl/AddressRepository.cs]: Izbaci file db provider

File: CSharpBasics/CSB.Repository/Impl/AddressRepository.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            var newId = _fileDb.Addresses.Max(x => x.Id);
+            address.Id = newId + 1;
+            _fileDb.Addresses.Add(address);
+            _fileDb.Save();
+            return address.Id;
+        }
+
+        public IReadOnlyList<Address> GetAddresses(int employeeId)
+        {
+            return _fileDb.Addresses.Where(x => x.EmployeeId == employeeId).ToList();
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Impl/AddressRepository.cs]: koristi dbContext.Addresses

File: CSharpBasics/CSB.Repository/Impl/AddressRepository.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+using CSB.Repository.Interfaces;
+
+namespace CSB.Repository.Impl
+{
+    internal class AddressRepository : IAddressRepository
+    {
+        private FileDbProvider _fileDb;
+
+        public int AddAddress(Address address)
+        {
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Impl/AddressRepository.cs:15]: koristi dbContext.Addresses

**Review by MilosSlavic — CHANGES_REQUESTED** [review]

**Comment by Excession27** [comment]
dbContext

File: CSharpBasics/CSB.Repository/Impl/AddressRepository.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    internal class AddressRepository : GenericRepository<Address>, IAddressRepository
+    {
+        public AddressRepository(CbsDbContext dbContext) : base(dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        private readonly CbsDbContext dbContext;
+
+        public int AddAddress(Address address)
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Impl/AddressRepository.cs]: Ukloni AddAddress

File: CSharpBasics/CSB.Repository/Entities/BaseEntity.cs [hunk 7]
```diff
@@ -3,6 +3,7 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using CSB.Repository.Entities;
```
Comment (MilosSlavic) [inline CSharpBasics/CSB.Repository/Entities/BaseEntity.cs:6]: Ukloni using

**Review by MilosSlavic — APPROVED** [review]

**Review by MilosSlavic — APPROVED** [review]

**Pull request merged**